packages/relay/src/lib/services/hapiService/hapiService.ts

```diff
@@ -0,0 +1,284 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import dotenv from 'dotenv';
+import findConfig from 'find-config';
+import { AccountId, Client, PrivateKey } from '@hashgraph/sdk';
+import { Logger } from 'pino';
+import { Registry, Counter } from 'prom-client';
+import { SDKClient } from '../../clients/sdkClient';
+import constants from '../../constants';
+import HbarLimit from '../../hbarlimiter';
+
+export default class HAPIService {
+  private transactionCount: number;
+  private errorCodes: number[];
+  private resetDuration: number;
+  private shouldReset: boolean;
+
+  private isReinitEnabled: boolean;
+  private isTimeResetDisabled: boolean;
+
+  private initialTransactionCount: number;
+  private initialErrorCodes: number[];
+  private initialResetDuration: number;
+
+  private hederaNetwork: string;
+  private clientMain: Client;
+
+  /**
+   * The SDK Client use for connecting to both the consensus nodes and mirror node. The account
+   * associated with this client will pay for all operations on the main network.
+   *
+   * @private
+   */
+  private client: SDKClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private logger: Logger;
+
+  /**
+   * This limiter tracks hbar expenses and limits.
+   * @private
+   */
+  private hbarLimiter: HbarLimit;
+  private readonly register: Registry;
+  private clientResetCounter: Counter;
+
+  /**
+   * @param {Logger} logger
+   * @param {Registry} register
+   */
+  constructor(logger: Logger, register: Registry, hbarLimiter: HbarLimit) {
```

Comment:
Maybe?
```suggestion
  constructor(logger: Logger, clientResetCounter: Counter, hbarLimiter: HbarLimit) {
```

Reply:
I think we need the Registry contrainer, because it's one instances passed around in the whole app, coming from relay class. This counter is something i've added. And it's not present in the relay class, where this object is initialized and relay class doesn't need to know about it.